Shapes/rml_rules_shape.ttl

```diff
@@ -0,0 +1,1683 @@
+    ns1:targetClass <http://www.wiwiss.fu-berlin.de/suhl/bizer/D2RQ/0.1#Database> .
+
+<http://schema.org/RRLogicalTable> a ns1:NodeShape ;
+    ns1:closed true ;
+    ns1:description """
+    Represents a logical table.
+    """ ;
+    ns1:ignoredProperties ( rdf:type ) ;
+    ns1:message """
+    rr:LogicalTable must provide either one rr:tableName or one rr:sqlQuery.
+    rr:sqlVersion is optional. 
+    """ ;
+    ns1:name "rr:LogicalTable" ;
+    ns1:property [ ns1:description """
+        Exactly one rr:tableName or one rr:sqlQuery is required.
+        """ ;
+            ns1:maxCount 1 ;
+            ns1:message """
+        Exactly one rr:tableName or one rr:sqlQuery is required.
+        """ ;
+            ns1:minCount 1 ;
+            ns1:name "rr:tableName/rr:sqlQuery" ;
+            ns1:path [ ns1:alternativePath ( <http://www.w3.org/ns/r2rml#tableName> <http://www.w3.org/ns/r2rml#sqlQuery> ) ] ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        Schema-qualified name of a table or view.
+        """ ;
+            ns1:message """
+        rr:tableName must be a string and may only be provided once.
+        """ ;
+            ns1:name "rr:tableName" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://www.w3.org/ns/r2rml#tableName> ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        A valid SQL query.
+        """ ;
+            ns1:message """
+        rr:sqlQuery must be a string and may only be provided once.
+        """ ;
+            ns1:name "rr:sqlQuery" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://www.w3.org/ns/r2rml#sqlQuery> ],
+        [ ns1:description """
+        An identifier for a SQL version.
+        """ ;
+            ns1:in ( <http://www.w3.org/ns/r2rml#SQL2008> ) ;
+            ns1:message """
+        rr:sqlVersion may only be provided once and must its value must be 
+        rr:SQL2008.
+        """ ;
+            ns1:name "rr:sqlVersion" ;
+            ns1:nodeKind ns1:IRI ;
+            ns1:path <http://www.w3.org/ns/r2rml#sqlVersion> ] .
+
+<http://schema.org/RRTriplesMapShape> a ns1:NodeShape ;
+    ns1:closed true ;
+    ns1:description """
+    Represents a triples map.
+    """ ;
+    ns1:ignoredProperties ( rdf:type ) ;
+    ns1:message """
+    rr:TriplesMap requires exactly one rr:subject or one rr:subjectMap and zero
+    or more rr:predicateObjectMaps.
+    """,
+        "rr:TriplesMap" ;
+    ns1:property [ ns1:description """
+        An IRI reference for use as subject for all the RDF triples generated 
+        from a logical table row or iterator.
+        """ ;
+            ns1:message """
+        rr:subject must be an IRI or blank node.
+        """ ;
+            ns1:name "rr:subject" ;
+            ns1:node <http://schema.org/RRsubjectShape> ;
+            ns1:nodeKind ns1:BlankNodeOrIRI ;
+            ns1:path <http://www.w3.org/ns/r2rml#subject> ],
```

Comment:
This would make the shapes more complex.
I tried to avoid as much as possible these statements because of that and have proper messages when a violation is found.